### PR TITLE
Migrate build settings to xcconfigs

### PR DESCRIPTION
**Pull request by saagarjha**
- [X] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [ ] This implements/fixes issue #.

---

**Description:**
Migrate build settings to xcconfigs, so they're out of the project file itself. In addition to reducing the chance of merge conflicts, this makes it far easier to create more complex build setups that we will need in the future.

### CODE REVIEW

File: iina/Shared.xcconfig [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+CLANG_ENABLE_OBJC_ARC = YES
+CLANG_ENABLE_OBJC_WEAK = YES
+CLANG_WARN__DUPLICATE_METHOD_MATCH = YES
+CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES
+CLANG_WARN_BOOL_CONVERSION = YES
+CLANG_WARN_COMMA = YES
+CLANG_WARN_CONSTANT_CONVERSION = YES
+CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES
+CLANG_WARN_DIRECT_OBJC_ISA_USAGE = YES_ERROR
+CLANG_WARN_DOCUMENTATION_COMMENTS = YES
```
Comment (low-batt) [inline iina/Shared.xcconfig:26]: I'm getting a bunch of new warnings such as "Not a Doxygen trailing comment". If I change this to "NO" the new warnings are gone.

Searching the iina.xcodeproj file in develop (not this PR) I see:
```text
low-batt@gag iina.xcodeproj (develop $=)$ grep COMMENT project.pbxproj 
				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
low-batt@gag iina.xcodeproj (develop $=)$ 
```

However searching for "comment" in the Xcode build setting using the UI I see all "no" and no "yes":
pr-4041

I'm confused as to where the "no" settings are coming from. But I wasn't getting these warnings before.

Comment (saagarjha) [reply iina/Shared.xcconfig:26]: The new settings don't precisely match the old ones, I stole them from a fresh project and then diffed them to get a set that would be appropriate for IINA (i.e. I picked the customized ones that we wanted, and reverted to Xcode's default for ones that were probably counterproductive). For warnings I tended towards enabling them. In this case, I guess we should either look at getting the comments to match up or temporarily disabling this setting.

Comment (low-batt) [reply iina/Shared.xcconfig:26]: From what I remember the offending comments are in FFmpeg include files. We'd have to get that project to change their sources, or apply patches each time we update FFmpeg.

I too prefer to enable most warnings. In some projects I've enabled "treat warnings as errors" as it was the only way to get some developers to stop committing sloppy code that introduced warnings.

**Comment by uiryuu** [comment]
Can we move the `xcconfig` files out of `iina/`? Or at least create a folder for them.

File: iina/Shared.xcconfig (hunk 1 above)
Comment (saagarjha) [reply iina/Shared.xcconfig]: >  We'd have to get that project to change their sources, or apply patches each time we update FFmpeg.

I'll look into this. Let's keep it in for now and if there isn't any progress we'll turn it off until they fix it.

> In some projects I've enabled "treat warnings as errors" as it was the only way to get some developers to stop committing sloppy code that introduced warnings.

I trust you all more than that :)

**Comment by saagarjha** [comment]
Sure, I just left it in there because that's where we've been sticking all the code, including things used across targets.

**Comment by saagarjha** [comment]
Screenshot 2022-11-07 at 17 06 41

One last time, huh…

**Comment by saagarjha** [comment]
Moved it out into a top-level Configs/ folder. Let's see what CI thinks…

**Pull request merged**